framework/DataObjects/XDataSet.py

```diff
@@ -853,6 +787,70 @@ def _convertFinalizedDataRealizationToDict(self,rlz):
         new[k] = v
     return new
 
+  def _convertToDict(self):
+    """
+      Casts this dataObject as dictionary.
+      @ In, None
+      @ Out, asDataset, xr.Dataset or dict, data in requested format
+    """
+    self.raiseAWarning('DataObject._convertToDict can be a slow operation and should be avoided where possible!')
+    # container for all necessary information
+    dataDict = {}
+    # supporting data
+    dataDict['dims']     = self.getDimensions()
+    dataDict['metadata'] = self.getMeta(general=True)
+    # main data
+    ## initialize with np arrays of objects
+    dataDict['data'] = dict((var,np.zeros(self.size,dtype=object)) for var in self.vars+self.indexes)
+    ## loop over realizations to get distinct values without NaNs
+    for var in self.vars:
+      # how we get and store variables depends on the dimensionality of the variable
+      dims=self.getDimensions(var)[var]
+      # if scalar (no dims and not an index), just grab the values
+      if len(dims)==0 and var not in self.indexes:
+        dataDict['data'][var] = self.asDataset()[var].values
+        continue
```

Comment:
do we need this line?

Reply:
Yes, it skips the rest of the code and goes back to the "for" loop.